### ISSUE TITLE
E2E Encryption for Photos

### ISSUE THREAD

**Issue by whatupcuzz84** [balzack/databag#26]
Is there any way to add E2E encryption for photos/videos? I did a test and once i added a photo in a topic I'm able to look on the server side and rename the file to filename.JPG and im able to see the photo.

![Screenshot 2023-04-22 144750](https://user-images.githubusercontent.com/53591018/233801450-080eed06-548d-4bd2-a747-22fb0de20c08.png)

**Comment by balzack** [comment]
I currently don't support sending photos/videos on e2e topics. The easiest approach would be to implement a download & decrypt step, but I would much prefer a streaming solution. I am currently looking into container formats that allow for an encrypted bitstream, but support for this feature is still a ways out. (sry)

Note: e2e is not enabled by default, you need to setup a 'seal' key in your profile, and then when you start the topic, you need to switch 'sealed channel' to on. So as to not be misleading, the 'photo' and 'video' icons disappear when communicating through a sealed channel.

**Comment by hnt3b453** [comment]
For me, long term, the lack of e2e on non-text items would be a deal breaker.

Short term, e2e on photos would be awesome, as the vast majority of non text items I send are photos, so that'll cover like 90% of my use case, and knowing videos aren't currently e2e, I'd just avoid sending them until you were able to implement it.

**Comment by balzack** [comment]
Sounds good. I'll have a solution for images in the next week or two.

As for video it's totally possible, I've just been dragging my feet hoping for the ideal solution. I could probably get videos under 1MB working in the short term, but large videos are the issue; I still hope to avoid the dowload/decrypt approach, but that may be my only option.

Thanks for your support :)

**Comment by hnt3b453** [comment]
Totally understandable, and fwiw, in my use case, I'm more than willing to wait for a good video solution. I don't really send a lot of video through chat. Thank you!!!

**Comment by whatupcuzz84** [comment]
Thanks for the reply. Yea I’ve learned to use the seal option just didn’t know if I was missing something with photos. I would love the feature of e2e encryption on any file typeOn Apr 22, 2023, at 2:58 PM, Pierre Balzack ***@***.***> wrote:﻿
I currently don't support sending photos/videos on e2e topics. The easiest approach would be to implement a download & decrypt step, but I would much prefer a streaming solution. I am currently looking into container formats that allow for an encrypted bitstream, but support for this feature is still a ways out. (sry)
Note: e2e is not enabled by default, you need to setup a 'seal' key in your profile, and then when you start the topic, you need to switch 'sealed channel' to on. So as to not be misleading, the 'photo' and 'video' icons disappear when communicating through a sealed channel.

**Comment by whatupcuzz84** [comment]
Yea but I’d also like to see only e2e encryption for pdfs and other document formats essentially if it sent through the app I’d love for it to be encrypted On Apr 22, 2023, at 7:56 PM, hnt3b453 ***@***.***> wrote:﻿
For me, long term, the lack of e2e on non-text items would be a deal breaker.
Short term, e2e on photos would be awesome, as the vast majority of non text items I send are photos, so that'll cover like 90% of my use case, and knowing videos aren't currently e2e, I'd just avoid sending them until you were able to implement it.

**Comment by balzack** [comment]
Content that is under 1 or 2MB should be pretty easy to include within e2e. I'll focus on images first, but other file types will follow shortly afterwards.

I had originally assumed that because there is no server side replication, that e2e would be less of an interest. As long as your account resides on a server you control then topics you start would never leave your control unencrypted.

This does gloss over the unexpected scenario of you server being compromised, at which point you would have wished your content was all e2e....or that you may communicate with someone that is not hosting their own account.

In short, I am increasing the priority of the e2e related stories in the backlog. Thanks for the feedback!

**Comment by balzack** [comment]
The mobile and browser apps are now updated to support image/audio/video media over sealed channels.

If you get a chance can you test it out and see if it meets your expectations. The loading time for sealed media is a little longer, so I scale the images to 1 megapixel. Unsealed channels remain unchanged

To test you will need to update the container (and mobile apps if you use them)

Thanks you.

**Comment by whatupcuzz84** [comment]
I’ll test now!

**Comment by whatupcuzz84** [comment]
Sorry busy saturday. im trying to update everything. im running a docker on unraid and there wasnt an update available. I forced an update and im trying to test now.

**Comment by balzack** [comment]
no rush, but thanks for doing this.

I haven't used unraid before, so I don't know how the docker images become available there. I need to start using it so I can understand that process.

**Comment by whatupcuzz84** [comment]
I’m sorry man I’ve been sick I should be able to actually test tomorrow. I got a docker update. Its built from your repo so its as close to official as it can be without you building.

**Comment by balzack** [comment]
Sounds good. No rush seriously, I'm just super happy you're testing it out. Feel better!

**Comment by balzack** [comment]
Awesome, thanks for verifying!
